nemo/collections/nlp/data/language_modeling/megatron/gpt_prompt_learning_dataset.py

```diff
@@ -0,0 +1,237 @@
+        text into token ids. Also replaces the <|VIRTUAL_PROMPT_#|> placeholders in 
+        the task templates with the actual virtual prompt token ids. 
+
+        params:
+            dataset: A list of json objects or a dictionary objects each
+                     containing the information needed for a training example
+        """
+        skipped = 0
+
+        for json_line in tqdm(dataset):
+
+            # Read example dict or load the information for a single example from .json file
+            if type(json_line) == dict:
+                doc = json_line
+            else:
+                doc = json.loads(json_line)
+
+            taskname = doc["taskname"]
+            prompt_template = self.task_templates[taskname]["prompt_template"]
+            prompt_template_fields = self.task_templates[taskname]["prompt_template_fields"]
+            prompt_token_splits = self.task_templates[taskname]["prompt_token_splits"]
+            input_example = prompt_template
+
+            assert (
+                sum(prompt_token_splits) == self.total_virtual_tokens
+            ), "Sum of prompt token split values must equal total number of prompt tokens"
+            assert prompt_template.count('<|VIRTUAL_PROMPT_') == len(
+                prompt_token_splits
+            ), "The number of '<|VIRTUAL_PROMPT_n|>' markers and the number of prompt token splits must match"
+
+            # Format the input example according to the template
+            for field in prompt_template_fields:
+                if field in doc.keys():
+                    field_text = doc[field]
+                    input_example = input_example.replace('{' + field + '}', field_text)
+
+                # If some fields from the template aren't present, e.g. {answer} during inference
+                # just remove that field from the template, leaving the space blank
+                else:
+                    input_example = input_example.replace('{' + field + '}', "")
+
+            total_inserted_tokens = 0
+
+            # Insert the correct number of pseudo tokens at the <|virtual_PROMPT_n|> markers
+            for idx in range(len(prompt_token_splits)):
+                split_start = total_inserted_tokens
+                split_end = total_inserted_tokens + prompt_token_splits[idx]
+                pseudo_tokens_for_split = "".join(self.pseudo_tokens[split_start:split_end])
+                input_example = input_example.replace(f'<|VIRTUAL_PROMPT_{idx}|>', pseudo_tokens_for_split)
+                total_inserted_tokens = split_end
+
+            input_ids = self.tokenizer.text_to_ids(input_example)
+
+            # Add BOS/EOS if desired, adds EOS by default
+            if self.add_bos:
+                input_ids = [self.tokenizer.bos_id] + input_ids
+            if self.add_eos:
+                input_ids = input_ids + [self.tokenizer.eos_id]
+
+            # Skip example if the final length doesn't fit length requirements
```

Comment:
can we have the option to truncate some of the input fields instead of skipping it? Otherwise, it might throw away a lot of data records.

Reply:
Sure, I'll add back your truncation logic